Scrapy-Cluster UI

damienkilgannon opened the pull request:
PR to merge my UI branch into IST UI branch. For discussions and collaboration. 

This code has been lingering around on my computer for a while now I think its about time I share it and try and get it to a place where it can be merged in. Further work still required on testing but core pieces of the ui_service are in place.

damienkilgannon commented:
Using Plotly for some simple visuals. Making stats requests every 15mins and caching up to 10 requests for building visuals. <img width="1673" alt="screen shot 2017-07-17 at 3 39 15 pm" src="https://user-images.githubusercontent.com/17731940/28274007-9e395072-6b07-11e7-97a8-65eb6bc543a0.png">

madisonb commented:
What is the difference between this PR and #95 ? Is this an improvement on the existing implementation and knocks off a number of items from the checklist at #116 ?

madisonb commented on ui/settings.py:
```diff
@@ -1,20 +1,24 @@
+STAT_REQ_FREQ = 910
+
+# REST host information
+REST_HOST = 'http://rest:5343'
```
Change this to `localhost`, I think the `docker/` settings should still be `rest`, but the travis unit tests that run vanilla centos or ubuntu look for `localhost` as the service endpoint (which is causing the Travis build to partially fail)

damienkilgannon commented:
@madisonb its an improvement on #95 and deals with several items from the checklist.

madisonb commented on run_online_tests.sh:
```diff
@@ -21,38 +21,18 @@ else
-if [ $? -eq 1 ]; then
-    echo "crawler tests failed"
-    exit 1
-fi
```
did you mean to commit this?

damienkilgannon replied:
no that was by mistake, have corrected it with latest commit

damienkilgannon commented:
@madisonb got the integration tests working via starting and stopping the rest service in the `./run_online_tests.py` script .... but now this other problem with the kafka_montior tests ... any ideas?

madisonb commented:
We need to ensure the docker tests are ran for the UI. I dont see any test output in the build log [here](https://travis-ci.org/istresearch/scrapy-cluster/jobs/255023295) near the bottom, nor do I see the container being used.

I think we need to update [this](https://git.io/vQhlg) line for travis to ensure the ui container has it's unit and integration test ran, the compose file is located [here](https://git.io/vQhl5) and it needs to have a UI service, except it should not be using a build context and instead reflect the container tag built by travis prior.

The build log for the normal ubuntu setup is also odd, but I presume that is because none of the other components are working (hence the "did not get response from kafka logs"). Given that we are only testing ui <-> rest, I think that is acceptable here.

damienkilgannon commented:
@madisonb fixed those issues in docker.sh and docker-compose.test.yml. Yeah, those warning messages in the build logs are because the kafka-monitor and redis-monitor are not running. Maybe all the services should be started in './run_online_tests.sh' for the integration tests regardless?

madisonb commented:
I think it is fine, let me take some time to spin the new UI up and I actually think this is pretty easy to merge in. Is it Python 3 compatible out of the box? I would like to hit the check box so the other pending PR can be merged in as well.

madisonb commented:
@damienkilgannon I have made a number of code updates in order to get the crawl submit job working, and while I cannot get the plotly graphs to update after a single data point, I have been able to get them all working and generating graphs. 

Please see [this](https://git.io/v7vmB) commit in particular for the updates I have made.

damienkilgannon commented:
@madisonb ok great .... i will checkout istresearch:ui now and have a look

damienkilgannon commented:
FYI: 15min data points graphed but timing is not very accurate due to the timestamp been set in the flow of the application.
<img width="1672" alt="screen shot 2017-07-20 at 12 05 05 pm" src="https://user-images.githubusercontent.com/17731940/28414526-f2d0e0f8-6d43-11e7-85f4-f38ea3d294b4.png">

madisonb commented:
So I don't want to prevent us from merging this code into the `dev` branch just because it doesnt have everything possible from a UI perspective. 

I think I would like to see more use of the [stats](https://scrapy-cluster.readthedocs.io/en/latest/topics/kafka-monitor/api.html#stats-api) API, in order to populate some of the various pages. Unit testing is probably what I would like to hit first (angular unit tests, or more tests for the base flask app).

Otherwise if you are good with this PR as is, we can add issues and work more PR's against that.

damienkilgannon commented:
@madisonb it wasn't my intention to try to merge to dev just yet, I had this code sitting in my repo and wanted to merge to istresearch:ui .... so we could do as you say, create tickets for additional features and the unittests and so forth. Or we can leave it as it is now and continue working on it there doesn't matter to me? I have been some changes made on the stats (using the sliding window stats correctly) which I will push latter today/tomorrow morning. What branch should I add those on to?

madisonb commented:
Correct I meant to say `ui` branch. Lets continue to make PR's against it and keep checking things off.

damienkilgannon commented:
@madisonb so i have fixed the way the UI service was reading the stats and added to the branch istresearch/scrapy-cluster:damienkilgannon-ui ... do we want to merge that branch into istresearch/scrapy-cluster:ui, close this PR and work on the branch? orrr?
